compiler/optimizer/SinkStores.cpp

```diff
@@ -1124,61 +1107,6 @@ void TR_SinkStores::lookForSinkableStores()
                   }
                }
 
-            // TR_MethodMetaData symbols are thread local and may be killed/used by any arbitrary symRef
-            else if (sinkMethodMetaDataStores() &&
```

Comment:
`sinkMethodMetaDataStores` is not mentioned in the PR and commit messages.

While I see how the other code involving it can be removed (because of the connection to `MetaDataList`), it's a little less clear to me why this code here is safe to remove. Could you provide some more details in a comment?

Reply:
Yes, but I think I got a little bit ahead of myself.  I think I can best answer your question with another commit added to this PR.  I'll mark this WIP until I've done that.